subsys/bluetooth/mesh/beacon.c

```diff
@@ -303,13 +303,6 @@ static void secure_beacon_recv(struct net_buf_simple *buf)
 
 	cache_add(data, sub);
 
-	/* If we have NetKey0 accept initiation only from it */
-	if (bt_mesh_subnet_get(BT_MESH_KEY_PRIMARY) &&
-	    sub->net_idx != BT_MESH_KEY_PRIMARY) {
-		BT_WARN("Ignoring secure beacon on non-primary subnet");
-		goto update_stats;
-	}
```

Comment:
I'm not sure removing this part is correct. Section 3.8.4 in the Mesh Profile specification 1.0.1 states the following:
"
If a device on a primary subnet receives an update on the primary subnet, it shall propagate the IV update to all other subnets. If a device on a primary subnet receives an IV update on any other subnet, the update shall be ignored.
"